src/bridge/cockpitdbusjson.c

```diff
@@ -788,25 +788,34 @@ static JsonObject *
 build_json_error (GError *error)
 {
   JsonObject *object;
-  JsonArray *reply;
-  JsonArray *args;
-  gchar *error_name;
 
   object = json_object_new ();
-  reply = json_array_new ();
-  args = json_array_new ();
 
-  error_name = g_dbus_error_get_remote_error (error);
-  g_dbus_error_strip_remote_error (error);
+  if (g_error_matches(error, G_IO_ERROR, G_IO_ERROR_TIMED_OUT))
+    {
+      json_object_set_string_member (object, "error", "timeout");
```

Comment:
Indeed. I did it this way because the documentation for `invocation.fail()` says that it returns a problem code in the `problem` field. According to `cockpit.js`, this field is only set when the `error` field is a string instead of an array.